roles/nova-data/tasks/libvirt.yml

```diff
@@ -56,29 +58,157 @@
+
+- name: create server cert
+  command: "certtool --generate-certificate
+           --template /tmp/server-cert-template
+           --load-privkey /etc/pki/libvirt/private/serverkey.pem
+           --load-ca-certificate /etc/pki/CA/cacert.pem
+           --load-ca-privkey /tmp/cacert.key
+           --outfile /etc/pki/libvirt/servercert.pem"
+  when: ca_install|changed or ca_key_install|changed or ca_template|changed
+  notify: restart libvirt-bin
+
+- name: upload client certificate template
+  template:
+    src: client-cert-template.j2
+    dest: /tmp/client-cert-template
+  register: client_template
+
+- name: create client private key
+  shell: "certtool --generate-privkey > /etc/pki/libvirt/private/clientkey.pem"
+  when: client_template|changed
+
+- name: create client cert
+  command: "certtool --generate-certificate
+           --template /tmp/client-cert-template
+           --load-privkey /etc/pki/libvirt/private/clientkey.pem
+           --load-ca-certificate /etc/pki/CA/cacert.pem
+           --load-ca-privkey /tmp/cacert.key
+           --outfile /etc/pki/libvirt/clientcert.pem"
+  when: client_template|changed
+  notify: restart libvirt-bin
+
+- name: set dir owner
+  file:
+    dest: /etc/pki/libvirt
+    recurse: yes
+    owner: root
+    group: kvm
+
+- name: set perms
+  file:
+    dest: "{{ item.value.path }}"
+    mode: "{{ item.value.mode }}"
+  with_dict:
+    1:
+      path: /etc/pki/libvirt
+      mode: "0755"
+    2:
+      path: /etc/pki/libvirt/private
+      mode: "0750"
+    3:
+      path: /etc/pki/libvirt/servercert.pem
+      mode: "0440"
+    4:
+      path: /etc/pki/libvirt/private/serverkey.pem
+      mode: "0440"
+    5:
+      path: /etc/pki/libvirt/clientcert.pem
+      mode: "0400"
+    6:
+      path: /etc/pki/libvirt/private/clientkey.pem
+      mode: "0400"
+
+- name: remove temporary files
+  file:
+    path: "{{ item }}"
+    state: absent
+  with_items:
+    - /tmp/cacert.key
+    - /tmp/server-cert-template
+    - /tmp/client-cert-template
+
 - name: update various lines in libvirtd.conf
-  lineinfile: dest=/etc/libvirt/libvirtd.conf regexp="{{ item.value.regexp }}"
-              line="{{ item.value.line }}"
+  lineinfile:
+    dest: /etc/libvirt/libvirtd.conf
+    regexp: "{{ item.value.regexp }}"
+    line: "{{ item.value.line }}"
   with_dict:
-    ltcp:
+    listen_tcp:
       regexp: '^listen_tcp\s*='
       line: "listen_tcp = 1"
-    ltls:
+    listen_tls:
       regexp: '^listen_tls\s*='
-      line: "listen_tls = 0"
-    atcp:
+      line: "listen_tls = 1"
+    auth_tcp:
       regexp: '^auth_tcp\s*='
       line: 'auth_tcp = \"none\"'
+    auth_tls:
+      regexp: '^auth_tls\s*='
+      line: 'auth_tls = \"none\"'
+    no_verify_cert:
+      regexp: '^tls_no_verify_certificate\s*='
+      line: 'tls_no_verify_certificate = 1'
+    no_verify_ip:
+      regexp: '^tls_no_verify_address\s*='
+      line: 'tls_no_verify_address = 1'
```

Comment:
`tls_no_verify_address` is for client certs address checking, and I thought it was definitely possible that this could come back to bite us, given the finicky nature of tls in libvirt.
https://avdv.github.io/libvirt/remote.html